### PR TITLE
Use netlink to set hairpin mode

### PR DESCRIPTION
**Pull request by LK4D4**
This should fix problem with hairpin on EL6. I have no EL6 to  try though.
/cc @phemmer

### CODE REVIEW

**Comment by phemmer** [comment]
I just spun up a VM last time when I was digging into the issue. I can again, but someone with real world usage might be better.

**Comment by LK4D4** [comment]
Actually I think this can be tested with `bridge` command from `iproute2`. Same netlink messages will be sent.

File: netlink/netlink_linux.go [hunk 1]
```diff
@@ -1253,25 +1252,33 @@ func SetMacAddress(name, addr string) error {
...
 	if enabled {
-		writeVal = []byte("1")
-	} else {
-		writeVal = []byte("0")
+		mode[0] = byte(1)
 	}
-	if _, err := sysFile.Write(writeVal); err != nil {
+
+	br := newRtAttr(syscall.IFLA_PROTINFO|syscall.NLA_F_NESTED, nil)
+	newRtAttrChild(br, 4, mode)
```
Comment (mrunalp) [inline netlink/netlink_linux.go]: The 4 here is IFLA_BRPORT_MODE, right? If so, could we define a constant for it?

Comment (LK4D4) [inline netlink/netlink_linux.go]: Yeah, of course.

**Comment by LK4D4** [comment]
@mrunalp I've added constant.

**Comment by mrunalp** [comment]
@LK4D4 thanks, I will test this out today.

**Comment by mrunalp** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**